vsphere/host_virtual_switch_structure.go

```diff
@@ -0,0 +1,151 @@
+func schemaHostVirtualSwitchBondBridge() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		// HostVirtualSwitchBeaconConfig
+		"beacon_interval": &schema.Schema{
+			Type:        schema.TypeInt,
+			Optional:    true,
+			Description: "Determines how often, in seconds, a beacon should be sent to probe for the validity of a link.",
+			Default:     1,
+		},
+
+		// LinkDiscoveryProtocolConfig
+		"link_discovery_operation": &schema.Schema{
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "Whether to advertise or listen for link discovery. Valid values are advertise, both, listen, and none.",
+			Default:     "listen",
+		},
+		"link_discovery_protocol": &schema.Schema{
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "The discovery protocol type. Valid values are cdp and lldp.",
+			Default:     "cdp",
+		},
+
+		// HostVirtualSwitchBondBridge
+		"network_adapters": &schema.Schema{
+			Type:        schema.TypeList,
+			Required:    true,
+			Description: "The link discovery protocol configuration for the virtual switch.",
+			Elem:        &schema.Schema{Type: schema.TypeString},
+		},
+	}
+}
+
+// expandHostVirtualSwitchBeaconConfig reads certain ResourceData keys and
+// returns a HostVirtualSwitchBeaconConfig.
+func expandHostVirtualSwitchBeaconConfig(d *schema.ResourceData) *types.HostVirtualSwitchBeaconConfig {
+	obj := &types.HostVirtualSwitchBeaconConfig{
+		Interval: int32(d.Get("beacon_interval").(int)),
+	}
+	return obj
+}
+
+// flattenHostVirtualSwitchBeaconConfig reads various fields from a
+// HostVirtualSwitchBeaconConfig into the passed in ResourceData.
+func flattenHostVirtualSwitchBeaconConfig(d *schema.ResourceData, obj *types.HostVirtualSwitchBeaconConfig) error {
+	d.Set("beacon_interval", obj.Interval)
+	return nil
+}
+
+// expandLinkDiscoveryProtocolConfig reads certain ResourceData keys and
+// returns a LinkDiscoveryProtocolConfig.
+func expandLinkDiscoveryProtocolConfig(d *schema.ResourceData) *types.LinkDiscoveryProtocolConfig {
+	obj := &types.LinkDiscoveryProtocolConfig{
+		Operation: d.Get("link_discovery_operation").(string),
+		Protocol:  d.Get("link_discovery_protocol").(string),
+	}
+	return obj
+}
+
+// flattenLinkDiscoveryProtocolConfig reads various fields from a
+// LinkDiscoveryProtocolConfig into the passed in ResourceData.
+func flattenLinkDiscoveryProtocolConfig(d *schema.ResourceData, obj *types.LinkDiscoveryProtocolConfig) error {
+	d.Set("link_discovery_operation", obj.Operation)
+	d.Set("link_discovery_protocol", obj.Protocol)
+	return nil
+}
+
+// expandHostVirtualSwitchBondBridge reads certain ResourceData keys and
+// returns a HostVirtualSwitchBondBridge.
+func expandHostVirtualSwitchBondBridge(d *schema.ResourceData) *types.HostVirtualSwitchBondBridge {
+	obj := &types.HostVirtualSwitchBondBridge{
+		NicDevice: sliceInterfacesToStrings(d.Get("network_adapters").([]interface{})),
```

Comment:
I'm trying to think of what might be good - `network_adapter_names` seems a little too redundant - they are not device IDs, but basically the friendly names of the NICs (`ie: vmnic0`). I personally think that documentation should be able to shore this up okay enough - but what are your thoughts?